src/clustering/clustering.jl

```diff
@@ -431,30 +443,30 @@ A vector of cluster assignments for each reef, ensuring that no cluster exceeds
 """
 function _constrained_kmeans_single_iteration(
     coordinates::Matrix{Float64},
-    k::Ref{Int},
-    k_spec::Int = 0,
+    k::Int64,
     max_cluster_size::Int64 = 6,
     max_split_distance::Int64 = 12000,
-    max_iter::Int64 = 1000,
+    max_iter::Int64 = 1000;
+    k_spec::Int = 0
 )::Vector{Int64}
-    clustering = kmeans(coordinates, k[]; maxiter=max_iter)
+    clustering = kmeans(coordinates, k; maxiter=max_iter)
     clustering_assignment::Vector{Int64} = copy(clustering.assignments)
```

Comment:
Why is the copy needed here? Does `clustering.assignments` get modified without it? I don't see `clustering_assignment` being modified below.